lib/facter/resolvers/linux/networking.rb

```diff
@@ -56,6 +58,24 @@ def parse_ip_command_line(line, mtu_and_indexes)
             mtu_and_indexes[name] = { index: index, mtu: mtu }
           end
 
+          def duplex(interface_name)
+            # not all interfaces support this, wifi for example does not
+            unless interface_name == 'lo'
+              @fact_list[:interfaces][interface_name][:duplex] = File.read("/sys/class/net/#{interface_name}/duplex").strip
```

Comment:
Is it safe to hardcode `lo` here? What about `lo:1`? Or should we be using a more general way of determining if the interface is loopback?

Can `@fact_list[:interfaces][interface_name]` be nil?

Can `"/sys/class/net/#{interface_name}/duplex"` not exist, if so `File.read` will raise.

Reply:
Alas, I can only answer one of these:

> Can `"/sys/class/net/#{interface_name}/duplex"` not exist, if so `File.read` will raise.

This file will always exist for each interface.  When it isn't relevant to the type of interface it is set to `unknown`.

Reply:
In theory I've got this sorted and have a check in place to avoid listing `lo:1` etc.